src/Backend/Synapse/Importer.php

```diff
@@ -105,29 +88,28 @@ public function importTable(
         return $this->importState->getResult();
     }
 
-    private function validateColumns(
+    private function getDestinationOptions(
         Storage\SourceInterface $source,
         Storage\Synapse\Table $destination
-    ): void {
-        if (count($source->getColumnsNames()) === 0) {
-            throw new Exception(
-                'No columns found in CSV file.',
-                Exception::NO_COLUMNS
-            );
-        }
-
-        $tableColumns = $this->sqlBuilder->getTableColumns(
+    ): DestinationTableOptions {
+        $tableRef = new SynapseTableReflection(
+            $this->connection,
             $destination->getSchema(),
             $destination->getTableName()
         );
 
-        $moreColumns = array_diff($source->getColumnsNames(), $tableColumns);
-        if (!empty($moreColumns)) {
-            throw new Exception(
-                'Columns doest not match. Non existing columns: ' . implode(', ', $moreColumns),
-                Exception::COLUMNS_COUNT_NOT_MATCH
-            );
+        $primaryKeysDefinition = DestinationTableOptions::PRIMARY_KEYS_DEFINITION_METADATA;
+        $primaryKeys = $source->getPrimaryKeysNames();
+        if ($primaryKeys === null) {
+            $primaryKeysDefinition = DestinationTableOptions::PRIMARY_KEYS_DEFINITION_DB;
+            $primaryKeys = $tableRef->getPrimaryKeysNames();
```

Comment:
Přemýšlám, že tu vyhodím exception ať nejde zapomenůt to nastavit.

Reply:
Toto tady v tomto PR nedává smysl ale dá smysl v tom následující. Jak nad tím přemýšlám tak to bude asi must have aby sa předešlo jakémukoliv fuckupu.

Reply:
Jako fuckup myslím, případ kdy uděláme přes optimalizovaný způsob Full load takže se vytvoří tabulka bez definovaných PK, a pak do ní uděláme inkrementální load, ale protože zjišťujeme PK z definice tabulky tak to nebude fungovat protože tam žádné nejsou.
Nevím jestli existuje případ, kdy má tabulka primární klíče ale neznáme je před importem, řekl bych že ne, buď jsou v metadatech nebo přišly z api jako parametr.